contracts/pcv/PCVGuardian.sol

```diff
@@ -0,0 +1,181 @@
+            for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+                _addWhitelistAddress(_whitelistAddresses[i]);
+            }
+        }
+    }
+
+    // ---------- Governor-Or-Guardian-Only State-Changing API ----------
+
+    /// @notice governor-or-guardian-only method to un-whitelist a pcvDesposit address to withdraw funds from
+    /// @param pcvDeposit the address to un-set as safe
+    function removeWhitelistAddress(address pcvDeposit)
+        external
+        override
+        onlyGuardianOrGovernor
+    {
+        _removeWhitelistAddress(pcvDeposit);
+    }
+
+    /// @notice batch version of unsetWhitelistAddress
+    /// @param _whitelistAddresses the addresses to un-whitelist
+    function removeWhitelistAddresses(address[] calldata _whitelistAddresses)
+        external
+        override
+        onlyGuardianOrGovernor
+    {
+        // improbable to ever overflow
+        unchecked {
+            require(_whitelistAddresses.length != 0, "empty");
+            for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+                _removeWhitelistAddress(_whitelistAddresses[i]);
+            }
+        }
+    }
+
+    /// @notice governor-or-guardian-or-pcv-guard method to withdraw funds from a pcv deposit, by calling the withdraw() method on it
+    /// @param pcvDeposit the address of the pcv deposit contract
+    /// @param amount the amount to withdraw
+    function withdrawToSafeAddress(address pcvDeposit, uint256 amount)
+        external
+        override
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.GUARDIAN,
+            TribeRoles.PCV_GUARD
+        )
+        onlyWhitelist(pcvDeposit)
+    {
+        _withdrawToSafeAddress(pcvDeposit, amount);
+    }
+
+    /// @notice governor-or-guardian-or-pcv-guard method to withdraw all at once funds from a pcv deposit, by calling the withdraw() method on it
+    /// @param pcvDeposit the address of the pcv deposit contract
+    function withdrawAllToSafeAddress(address pcvDeposit)
+        external
+        override
+        hasAnyOfThreeRoles(
+            TribeRoles.GOVERNOR,
+            TribeRoles.GUARDIAN,
+            TribeRoles.PCV_GUARD
+        )
+        onlyWhitelist(pcvDeposit)
+    {
+        _withdrawToSafeAddress(pcvDeposit, IPCVDeposit(pcvDeposit).balance());
+    }
+
+    // ---------- Internal Functions ----------
+
+    function _withdrawToSafeAddress(address pcvDeposit, uint256 amount)
+        internal
+    {
+        if (pcvDeposit._paused()) {
+            pcvDeposit._unpause();
+            IPCVDeposit(pcvDeposit).withdraw(safeAddress, amount);
+            pcvDeposit._pause();
+        } else {
+            IPCVDeposit(pcvDeposit).withdraw(safeAddress, amount);
+        }
+
+        emit PCVGuardianWithdrawal(pcvDeposit, amount);
+    }
+
+    function _addWhitelistAddress(address pcvDeposit) internal {
+        require(whitelistAddresses.add(pcvDeposit), "set");
```

Comment:
since you renamed these functions, you should probably rename the reverts as well